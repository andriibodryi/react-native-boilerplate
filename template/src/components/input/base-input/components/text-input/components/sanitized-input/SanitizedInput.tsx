import React, { useEffect, useState } from 'react';
import { TextInput } from 'react-native';

import { SanitizedInputTypes } from './sanitized-input.types';
import { defaultSanitizer } from '../../utils';

const SanitizedInput: React.FC<SanitizedInputTypes> = ({
  value,
  sanitizer = defaultSanitizer,
  onChangeText,
  maxLength,
  forceValueUpdate = false,
  ...rest
}) => {
  const [sanitized, setSanitized] = useState(value);

  useEffect(() => {
    if (forceValueUpdate) {
      setSanitized(sanitizer(value || ''));
    }
  }, [forceValueUpdate, sanitizer, value]);

  return (
    <TextInput
      onChangeText={v => {
        const sanitizedValue = sanitizer(v);
        setSanitized(() => sanitizedValue);
        onChangeText?.(sanitizedValue);
      }}
      value={sanitized}
      maxLength={maxLength}
      {...rest}
    />
  );
};

export default SanitizedInput;
