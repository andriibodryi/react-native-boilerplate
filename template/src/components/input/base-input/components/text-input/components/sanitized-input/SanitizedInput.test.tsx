import React from 'react';
import { render } from '@testing-library/react-native';

import SanitizedInput from './SanitizedInput';

describe('SanitizedInput', () => {
  it('should render correctly', () => {
    const tree = render(<SanitizedInput />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
