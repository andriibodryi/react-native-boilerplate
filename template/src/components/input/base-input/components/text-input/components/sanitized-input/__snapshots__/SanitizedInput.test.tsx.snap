// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SanitizedInput should render correctly 1`] = `
<TextInput
  allowFontScaling={true}
  onChangeText={[Function]}
  rejectResponderTermination={true}
  underlineColorAndroid="transparent"
/>
`;
